dodiscover/context_builder.py

```diff
@@ -0,0 +1,140 @@
+from typing import Optional, Set, Union
+
+import networkx as nx
+import pandas as pd
+
+from ._protocol import Graph
+from .context import Context
+
+
+class ContextBuilder:
+    """A builder class for creating Context objects ergonomically.
+
+    The context builder provides a way to capture assumptions, domain knowledge, and data.
+
+    Parameters
+    ----------
+    data : pd.DataFrame
+        A dataset in data-frame form, consisting of samples as rows and variables as columns.
+    """
+
+    _graph: Optional[Graph] = None
+    _included_edges: Optional[Union[nx.Graph, nx.DiGraph]] = None
+    _excluded_edges: Optional[Union[nx.Graph, nx.DiGraph]] = None
+    _observed_variables: Optional[Set[str]] = None
```

Comment:
```suggestion
    _observed_variables: Optional[Set[Column]] = None
```

In `dodiscover.typing.py`, there is a definition for the typing of variables that we can store.